### PR TITLE
tests/{sys_fido2_ctap/usbus_board_reset}: fix stdio_usb_serial_jtag dependency

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR fixes the `stdio_usb_serial_jtag` dependency problem for `tests/sys_fido2_ctap` and `usbus_board_reset`.

- There are boards that select the STDIO backend used depending on whether `usbus` is enabled. Usually the `fido2_ctap_transport_hid` module pulls in `usbus_hid` and thus `usbus`, but since this dependency resolution is done after reading the `Makefile.dep` of the board, it may happen that the wrong STDIO backend is selected. Therefore `usbus` is selected directly in the `Makefile` of `tests/sys_fido2_ctap` .
- To improve the selection of the `stdio_usb_serial_jtag` backend in `esp32s3-pros3`, it checks for any `usbus_%` module not only `usbus`.

### Testing procedure

GreenCI

### CODE REVIEW

File: tests/sys_fido2_ctap/Makefile [hunk 1]
```diff
@@ -4,6 +4,7 @@ BOARD ?= nrf52840dk
 include ../Makefile.tests_common
 
 USEMODULE += fido2_ctap_transport_hid
+#USEMODULE += usbus
```
[the remark this answers is not in the document]
Comment (gschorcht) [reply tests/sys_fido2_ctap/Makefile:7]: Oops, Good catch, I just forgot to remove the `#` after testing something different.

[the remark this answers is not in the document]
Comment (gschorcht) [reply tests/sys_fido2_ctap/Makefile]: Fixed.

**Comment by gschorcht** [comment]
> The reason for this mismatch is that nothing actually enables the `MODULE_USBUS`, `MODULE_USBUS_CDC_ACM` and `MODULE_USB_BOARD_RESET`. You could directly select them I suppose from the board but this would assume there would always be this as the STDIO.

Are you sure? `tests/sys_fido2_ctap/app.config.test`  enables `MODULE_USBUS` and we have the conditional in `SDTIO_IMPLEMENTATION`
```
  choice STDIO_IMPLEMENTATION
        bool "STDIO implementation"
        default MODULE_STDIO_CDC_ACM(=n) if MODULE_USBUS(=y)
        default MODULE_STDIO_TINYUSB_CDC_ACM(=n) if MODULE_TINYUSB_DEVICE(=n)
        default MODULE_STDIO_USB_SERIAL_JTAG(=n) if !MODULE_USBUS(=y) && !MODULE_TINYUSB_DEVICE(=n)
        depends on TEST_KCONFIG(=y)
```
which should select `MODULE_STDIO_CDC_ACM` because of `MODULE_USBUS(=y)`. If I open the symbol info for `MODULE_STDIO_USB_SERIAL_JTAG` I get the information that the dependencies are fulfilled.
```
Direct dependencies (=y):
     TEST_KCONFIG(=y)
  && CPU_FAM_ESP32C3(=n) || CPU_FAM_ESP32S3(=y)  (=y)
  && <choice STDIO_IMPLEMENTATION>
```
`MODULE_USB_BOARD_RESET` is another thing. I'm just wondering why `MODULE_STDIO_CDC_ACM` isn't selected.

**Comment by gschorcht** [comment]
Ok, the last CI run was successful. What next? Should I remove the Murdock settings, or should we try a complete nightly build or should we just try to merge and see what the nightly build says tomorrow?

**Comment by maribu** [comment]
merge now and cross fingers :)

**Review by maribu — APPROVED** [review]
please type the magic words when the commit affecting .murdock is dropped :)

**Comment by gschorcht** [comment]
bors merge

**Comment by gschorcht** [comment]
bors merge

**Pull request merged**